pkgs/tools/filesystems/irods/default.nix

```
@@ -1,27 +1,29 @@
-{ lib, stdenv, fetchFromGitHub, bzip2, zlib, autoconf, automake, cmake, gnumake, help2man , texinfo, libtool , cppzmq , libarchive, avro-cpp_llvm, boost, jansson, zeromq, openssl , pam, libiodbc, libkrb5, gcc, libcxx, which, catch2 }:
+{ lib, stdenv, fetchFromGitHub, bzip2, zlib, autoconf, automake, cmake, gnumake, help2man , texinfo, libtool , cppzmq , libarchive, avro-cpp_llvm, boost, jansson, zeromq, openssl , pam, libiodbc, libkrb5, gcc, libcxx, which, catch2, nanodbc_llvm, fmt, nlohmann_json, spdlog }:
 
 let
   avro-cpp=avro-cpp_llvm;
+  nanodbc=nanodbc_llvm;
 in
 let
   common = import ./common.nix {
```

This should use callPackage.

---

If you do the following you only need to inherit variables which are not defined in top-level.
```suggestion
  common = callPackage ./common.nix {
```

---

This fails with something I'm not enough skilled to understand:
```
[bzizou@bart:~/git/nixpkgs]$ nix-build . -A irods-icommands --cores 8 --show-trace
error: while evaluating the attribute 'buildInputs' of the derivation 'irods-icommands-4.2.11' at /home/bzizou/git/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:205:7:
while evaluating the attribute 'override' of the derivation 'irods-4.2.11' at /home/bzizou/git/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:205:7:
cannot coerce a set to a string, at /home/bzizou/git/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:205:7
```

---

It would only work if it would be a mkDerivation I think.